device/iot-device-client/src/test/java/tests/unit/com/microsoft/azure/sdk/iot/device/DeviceClientTest.java

```diff
@@ -147,6 +148,34 @@ public void constructorSuccess() throws URISyntaxException, IOException
         };
     }
 
+    @Test
+    public void constructorWithClientOptionsSuccess() throws URISyntaxException, IOException
+    {
+        // arrange
+        final String connString = "HostName=iothub.device.com;CredentialType=SharedAccessKey;CredentialScope=Device;deviceId=testdevice;SharedAccessKey=adjkl234j52=;";
+        final IotHubClientProtocol protocol = IotHubClientProtocol.MQTT;
+        final ClientOptions clientOptions = new ClientOptions();
+        clientOptions.setModelId("dtmi:Company:device:1");
+
+        // act
+        final DeviceClient client = new DeviceClient(connString, protocol, clientOptions);
+
+        // assert
+        new Verifications()
+        {
+            {
+                IotHubConnectionString iotHubConnectionString = Deencapsulation.newInstance(IotHubConnectionString.class, connString);
+                times = 1;
+
+                IoTHubConnectionType ioTHubConnectionType = Deencapsulation.getField(client, "ioTHubConnectionType");
+                assertEquals(SINGLE_CLIENT, ioTHubConnectionType);
+
+                TransportClient transportClient = Deencapsulation.getField(client, "transportClient");
```

Comment:
how is this assertion relevant here?

Reply:
From our legacy tests:  The constructor shall set the transportClient to null.

What do you think is releavant?

Reply:
I don't understand why initializing a device client sets the transportClient to null, and how that is relevant to this test case.